backend/src/modules/task/controllers/task.controller.ts

```diff
@@ -15,7 +28,28 @@ export class TaskController {
 
   @Post()
   async create(@Req() req: AuthenticatedRequest, @Body() dto: CreateTaskDto) {
-    const user = req.user; 
+    const user = req.user;
     return await this.taskService.createTask({ ...dto, createdBy: user.id });
   }
-}
+
+  @Get()
+  async getAll() {
+    return await this.taskService.getAllTasks();
+  }
+
+  @Patch(':id')
+  async editTask(
+    @Param('id', ParseIntPipe) id: number,
+    @Body() dto: EditTaskDto
+  ) {
+    return await this.taskService.editTask(id, dto);
+  }
+
+  @Delete(':id')
+  async deleteTask(
+    @Param('id', ParseIntPipe) id: number,
+    @Body() dto: DeleteTaskDto
```

Comment:
since this is delete request, you do not have to receive body, thus no need for any dto neither, so please remove them, and just accept id, and soft delete it if it exists